bot.py

```diff
@@ -338,6 +338,147 @@ async def display_current_list_of_summoners(ctx):
         await ctx.send(embed=create_embed(embed_data))
 
 
+# pylint: disable=too-many-locals, too-many-branches, too-many-statements
+@bot.command(name="remove", help="Remove player(s) from the list")
+async def remove_summoner(ctx, *, message):
+    """Remove summoner(s) from list
+    and send  the list to the bot"""
+
+    try:
+        # typing indicator
+        async with ctx.typing():
+            await asyncio.sleep(1)
+
+        # converting the message into list of summoners
+        summoner_to_remove_input = message.split(",")
+
+        # Exception case: attempt to remove more than 10 players
+        if len(summoner_to_remove_input) > MAX_NUM_PLAYERS_TEAM:
+            raise Exception (
+                "Limit Exceeded",
+                "You have exceeded a limit of 10 summoners! \
+                \nPlease remove {0} less summoners!".format(
+                    MAX_NUM_PLAYERS_TEAM - len(summoner_to_remove_input)
+                )
+            )
+
+        # Exception case: data/data.json file does not exist
+        if not os.path.exists(json_path):
+            raise Exception (
+                "Limit Exceeded",
+                "There is no summoner(s) added in the game.\nPlease add summoner(s) first!"
+            )
+
+        # for importing data from json file
+        file_data = ""
+        # initializing server id to a variable
+        server_id = str(ctx.guild.id)
+
+        if os.path.getsize(json_path) > 0:
+            with open(json_path, "r") as file:
+                file_data = json.load(file)
+        else:
+            raise Exception (
+                "Limit Exceeded",
+                "There is no summoner(s) added in the game.\nPlease add summoner(s) first!"
+            )
+
+        players_to_remove = []
+        summoners_to_remove = []
+        if server_id in file_data:
+            for player_name in summoner_to_remove_input:
+                for player in file_data[server_id]:
+                    if normalize_name(player_name) == normalize_name(player["user_name"]):
+                        players_to_remove.append(player)
+                        summoners_to_remove.append(player_name)
+            for player in players_to_remove:
+                file_data[server_id].remove(player)
```

Comment:
you can't do this while looping?

Reply:
It can be done with while loop. But why not for loop? What's the benefit?

Reply:
oh no Im not asking to use `while` loop. I'm asking if `file_data[server_id].remove(player)` can be done while looping the names before. Asking to refactor if possible. @jung1411 do you have any idea? 
We have two while-loop within each other + 2 other for loops that looks to be sooo inefficient for simple job.

Reply:
```
players_to_remove = []
if server_id in file_data:
    for player_name in summoner_to_remove_input:
        for player in file_data[server_id]:
            if normalize_name(player_name) == normalize_name(player["user_name"]):
                players_to_remove.append(player)
                file_data[server_id].remove(player)
                summoner_to_remove_input.remove(player_name)
```
If we do this, we can get rid of two loops and not use `summoners_to_remove`

Reply:
I think we can also remove `players_to_remove` since we are using it one time.
```
if server_id in file_data:
    for player_name in summoner_to_remove_input:
        for player in file_data[server_id]:
            if normalize_name(player_name) == normalize_name(player["user_name"]):
                file_data[server_id].remove(player)
                summoner_to_remove_input.remove(player_name)
```
I tried this in a notebook and it worked on my end, but see if this works on your end.

Reply:
@jung1411 That's going to break for-loop.
for loop is running for all player in `file_data[server_id]` and if we do `file_data[server_id].remove(player)`, it breaks the for-loop and loop will exit without looping other ones.

I have a feeling that using while loop would be more efficient in this case.
I will try to come up with something later on.

Reply:
```
summoner_to_remove = []
if server_id in file_data:
    for player_name in summoner_to_remove_input:
        for player in file_data[server_id]:
            if normalize_name(player_name) == normalize_name(player["user_name"]):
                file_data[server_id].remove(player)
                summoner_to_remove.append(player_name)

summoner_to_remove_input = list(set(summoner_to_remove_input) - set(summoner_to_remove))
```
This one works
![image](https://user-images.githubusercontent.com/46850644/125393508-143b2a80-e376-11eb-9eb2-c14936d486f0.png)

Reply:
issue handled by using pydash.find() function